block/retriever.go

```diff
@@ -19,10 +19,14 @@ func (m *Manager) ApplyBatchFromSL(slBatch *settlement.Batch) error {
 
 	m.logger.Debug("retrieved batches", "n", len(batchResp.Batches), "daHeight", slBatch.MetaData.DA.Height)
 
+	// if batch blocks have already been applied skip, otherwise it will fail in endheight validation (it can happen when syncing from blocksync in parallel).
```

Comment:
in this case, it should be called after the mutex lock

Reply:
good catch. moved after lock.